streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/system/controller/PassportController.java

```diff
@@ -102,6 +110,54 @@ public RestResponse signin(
         return new RestResponse().data(userInfo);
     }
 
+    @PostMapping("signinbycasdoor")
+    public RestResponse signinByCasdoor(@NotBlank(message = "{required}") String code,
+                                        @NotBlank(message = "{required}") String state) {
+        String token = "";
+        String username = "";
+        String password = "";
+        token = casdoorAuthService.getOAuthToken(code, state);
+        CasdoorUser casdoorUser = casdoorAuthService.parseJwtToken(token);
+        User user = userService.findByName(casdoorUser.getName());
+        if (user == null) {
+            User user2 = new User();
```

Comment:
It is not recommended to use the variable name "user2", because it does not provide a clearly meaning

Reply:
ok